src/gui/windows/multiplayer.zig

```diff
@@ -85,6 +86,46 @@ fn copyIp(_: usize) void {
 	main.Window.setClipboardString(ipAddress);
 }
 
+fn isCorrectNetAddress(str: []const u8) bool {
+	// min doamin name length is 2
+	if(str.len < 2) return false;
+
+	// check if port is valid
+	const portLess = blk: {
+		var iter = std.mem.splitScalar(u8, str, ':');
+		break :blk iter.first();
+	};
+	if(str.len > portLess.len) {
+		const port = str[portLess.len + 1 ..];
```

Comment:
Cubyz ports are allowed to start with a `?` which signals a symmetric NAT connection.